packages/components/src/components/StepIndicator/StepIndicatorSegment.tsx

```diff
@@ -0,0 +1,40 @@
+import React from "react";
+import { VisuallyHidden } from "ariakit/visually-hidden";
+import { Box } from "../../primitives/Box";
+
+export interface StepIndicatorSegmentProps
+  extends Omit<React.HTMLAttributes<HTMLOListElement>, "color"> {
+  /** The accessible text content of the step. */
+  children: string;
+  /** The current step that shows the user’s position. Inherited from StepIndicator. */
+  current?: number;
+  /** The index within the array of segments. Inherited from StepIndicator. */
+  index?: number;
+}
+
+/** The step indicator segment indicates the current step within the StepIndicator. */
+const StepIndicatorSegment = React.forwardRef<
+  HTMLLIElement,
+  StepIndicatorSegmentProps
+>(({ children, current, index = 1 }, ref) => {
+  return (
+    <Box.li
+      aria-current={current === index + 1}
```

Comment:
Why do we need to add 1 to the index here? Does index start from 0 and current doesn't? Should we make both start from 0 to avoid confusion.

Reply:
Yeah. Since an array index starts with 0. I'm ok with going with 0, so I added a note to the `current` prop description about it.